packages/valory/skills/abstract_round_abci/abci_app_chain.py

```diff
@@ -190,11 +190,11 @@ def get_paths(
                     accumulated_post_conditions,
                 )
                 if len(diff) != 0:
-                    raise ValueError(
+                    raise ValueError(  # pragma: no cover
```

Comment:
Please open an issue so we add tests for this and the below & above one

Reply:
This function the aforementioned issue perfectly. It is 250 lines, even if we subtract the nested function it's still well over 200 lines of code; and it contains numerous functionalities. It becomes unnecessarily difficult test and reason about, and should be refactored first, not after overly complicates tests are written to cover new nested functionalities that otherwise cannot be covered.

Reply:
https://github.com/valory-xyz/open-autonomy/issues/1692